sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -418,12 +398,17 @@ export class MessageSession extends LinkEntity {
           this.name,
           options
         );
-        if (!this._context.messageSessions[this.sessionId!]) {
-          this._context.messageSessions[this.sessionId!] = this;
+        if (
+          receivedSessionId != undefined &&
+          (this.sessionId === undefined || this.sessionId === receivedSessionId)
```

Comment:
Not very sure if we absolutely have to make this change because we dont really pass `null` around, but we do have these checks elsewhere, so lets maintain the pattern

```suggestion
          (this.sessionId == undefined || this.sessionId === receivedSessionId)
```